server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -272,6 +296,64 @@ public void onFailure(Exception e) {
         }
     }
 
+    static Map<String, IndexTemplateV2.DataStreamTemplate> resolveAutoCreateDataStreams(Metadata metadata,
+                                                                                        Set<String> autoCreateIndices) {
+        Map<String, IndexTemplateV2.DataStreamTemplate> autoCreateDataStreams = new HashMap<>();
+        Iterator<String> autoCreateIndicesIterator = autoCreateIndices.iterator();
+        while (autoCreateIndicesIterator.hasNext()) {
+            String indexName = autoCreateIndicesIterator.next();
+            String v2Template = MetadataIndexTemplateService.findV2Template(metadata, indexName, false);
```

Comment:
I think this means that if you have both v1 and v2 templates matching the same pattern and the v2 one is a data stream, it takes precedence even when `prefer_v2_templates=false`. I think this will cause some confusion and I would prefer to stick to respecting the option.